services/identity/service.go

```diff
@@ -143,35 +161,45 @@ func (s *Service) ProposeFetch(si *network.ServerIdentity, cnc *ProposeFetch) (n
 // that the voter is in the latest config.
 // An empty signature signifies that the vote has been rejected.
 func (s *Service) ProposeVote(si *network.ServerIdentity, v *ProposeVote) (network.Body, error) {
+	log.Lvl2(s, "Voting on proposal")
 	// First verify if the signature is legitimate
 	sid := s.getIdentityStorage(v.ID)
 	if sid == nil {
 		return nil, errors.New("Didn't find identity")
 	}
-	sid.Lock()
-	log.Lvl3("Voting on", sid.Proposed.Device)
-	owner, ok := sid.Latest.Device[v.Signer]
-	if !ok {
-		return nil, errors.New("Didn't find signer")
-	}
-	if sid.Proposed == nil {
-		return nil, errors.New("No proposed block")
-	}
-	hash, err := sid.Proposed.Hash()
-	if err != nil {
-		return nil, errors.New("Couldn't get hash")
-	}
-	if _, exists := sid.Votes[v.Signer]; exists {
-		return nil, errors.New("Already voted for that block")
-	}
-	log.Lvl3(v.Signer, "voted", v.Signature)
-	if v.Signature != nil {
-		err = crypto.VerifySchnorr(network.Suite, owner.Point, hash, *v.Signature)
+
+	// Putting this in a function because of the lock which needs to be held
+	// over all calls that might return an error.
```

Comment:
Because if there is an error, I need to be able to return from the function. As there are a lot of returns, I'd have to write `sid.Unlock` every time, because I also need it to be unlocked later. Perhaps it would be more clear to have the `Lock` before the function and the `Unlock` after the function?